contracts/bridge/contracts/common/message/OutboundLane.sol

```diff
@@ -245,6 +248,21 @@ contract OutboundLane is IOutboundLane, MessageVerifier, TargetChain, SourceChai
         }
     }
 
+    function on_messages_delivered(DeliveredMessages memory confirmed_messages) internal {
+        for (uint64 nonce = confirmed_messages.begin; nonce <= confirmed_messages.end; nonce ++) {
+            uint256 offset = nonce - confirmed_messages.begin;
+            bool dispatch_result = ((confirmed_messages.dispatch_results >> offset) & 1) > 0;
+            address submitter = messages[nonce].payload.sourceAccount;
+            bytes memory deliveredCallbackData = abi.encodeWithSelector(
+                IOnMessageDelivered.on_messages_delivered.selector,
+                nonce,
+                dispatch_result
+            );
+            (bool ok,) = submitter.call{value: 0, gas: MAX_GAS_PER_MESSAGE}(deliveredCallbackData);
```

Comment:
Require carefully audit for this call back, what is `IOnMessageDelivered.on_messages_delivered.selector`?

Reply:
The callback of `on_messages_delivered` is to let app layer aware the message dispatch result. 
https://github.com/paritytech/parity-bridges-common/blob/3651f4f90915fabaaa6c5fe160fdd26a0f23012f/modules/messages/src/lib.rs#L596
We have two ways to deal with it.
1. Message layer just save the dispatch result, and let app fetch the dispatch result.
2. Message layer pull the dispatch result to app layer.

Reply:
App need some callback to be triggered when message get confirmed, otherwise they will have to be trigged again by user manually, and also will break the transactional.

Current implementation looks good if apps's `IOnMessageDelivered` have permission control, the msg.sender here for this call is this OutboundLane right?

Reply:
And if submitter here is an EOA address, what will happen here?

Reply:
Yes, when do the call, the `msg.sender` will be `outboundlane`

Reply:
If submitter is an EOA address, the call is just like general transfer, but the value is 0.